uaclient/apt.py

```diff
@@ -653,3 +653,42 @@ def update_esm_caches(cfg) -> None:
     # in tests - down the rabbit hole, not worth it
     except apt.cache.FetchFailedException:
         logging.warning("Failed to fetch the ESM Apt Cache")
+
+
+def _get_apt_config():
+    import apt_pkg
+
+    # We need to clear the config values in case another module
+    # has already initiated it
+    for key in apt_pkg.config.keys():
+        apt_pkg.config.clear(key)
```

Comment:
This scares me a bit - maybe because I don't fully understand.

Would it be better or worse if we did something like:
```python
@lru_cache(maxsize=None)
def get_init_apt_pkg():
    import apt_pkg
    apt_pkg.init()
    return apt_pkg
```

and then use that function in the places we need to import apt_pkg and init it?


(and maybe more of a stretch, is there any way to use python3-apt in a non-global-state-ish way?)

Reply:
And I'm not sure how my suggestion would interact with our other usage of `import apt; apt.Cache()` :/

and I'm not sure how it would interact with other programs that use apt_pkg and are using our api :(

Reply:
Reading the docs here: https://apt-team.pages.debian.net/python-apt/library/apt_pkg.html#configuration-and-command-line-parsing

It _might_ be possible to do something like:
```python
def our_apt_related_fn():
    backup_apt_pkg_config = apt_pkg.config
    apt_pkg.config = apt_pkg.Configuration()
    apt_pkg.init_config()
    # do whatever we need to do with our special config
    # but before we return, restore the backup config
    apt_pkg.config = backup_apt_pkg_config
```

CC @renanrodrigo for your thoughts here as well

Reply:
Clearing all the config and `init`ing again is fine. Let's keep it as is.
IN THE FUTURE we will have a CtxMgr for our own cache, and then we can just believe apt_pkg without calling `.init` again.